src/shared/api/ComparisonGroupClient.ts

```diff
@@ -27,19 +26,19 @@ export type Group = {
 export type GroupData = Omit<VirtualStudyData, 'studyViewFilter'>;
 
 export default class ComparisonGroupClient {
-    // edge case: user deletes a group, then opens the panel again,
-    //          and the getGroups request responds before the deleteGroup
+    // edge case: user deletes/add a group, then opens the panel again,
```

Comment:
doesn't feel quite right to deal with this at client level.  our clients thus far don't have any such state (as far as I know).   i would deal with this in the store mobxpromises.  @adamabeshouse what do you think?

Reply:
@alisman I disagree, I think it's an issue the client should hide, too nitty gritty for every application to handle and rehandle

Reply:
@kalletlak i'm gonna go with my gut here.  when i make a request to a client, i am not expecting that request to be dependent on status of other requests.   we do not want to start cluttering our clients with this kind of business logic.   this belongs in the store.

Reply:
@kalletlak are we going to do this (move it to store)?

Reply:
moved.